### PR TITLE
Separate weak-algorithm logic into utility and private method.

### PR DESCRIPTION
**Pull request by cobratbq**
Extract PGP signature-specific logic into utility method for
PGPSignatureUtils. Move remaining control flow logic into separate
private method dedicated to determining next action after weak algorithm
is discovered. (Logging with desired log level and then determining
whether or not to abort.)

### CODE REVIEW

File: src/main/java/org/simplify4u/plugins/utils/PGPSignatureUtils.java [hunk 1]
```diff
@@ -40,6 +41,38 @@ private PGPSignatureUtils() {
         // No need to instantiate utility class.
     }
 
+    /**
+     * Check PGP signature for bad algorithms.
+     *
+     * @param signature PGP signature instance
+     * @return Returns null if no bad algorithms used, or algorithm name if used.
+     */
+    public static String checkWeakHashAlgorithm(PGPSignature signature) {
```
Comment (slawekjaranowski) [inline src/main/java/org/simplify4u/plugins/utils/PGPSignatureUtils.java:50]: for it is easy to write unit test with mocked signature

Comment (cobratbq) [reply src/main/java/org/simplify4u/plugins/utils/PGPSignatureUtils.java:50]: ah, *that* was the reason why I had not created the PR yet. The unit test was missing :-P
I've been making changes for 4 or 5 different goals, so got things mixed up.

You're right of course. I'll write a few unit tests.

**Comment by cobratbq** [comment]
@slawekjaranowski now for a different question ... was _SHA-1_ ever considered for the weak algorithm list? ... I noticed it is missing, but I would probably leave it out as I suspect many PGP public keys in active use still rely on it.

I'm not sure and I haven't investigated it yet, but I'm curious to hear if you thought about this already.

File: src/test/java/org/simplify4u/plugins/utils/PGPSignatureUtilsTest.java [hunk 2]
```diff
@@ -31,4 +37,37 @@ public void testLoadSignatureContentIsSignature() throws IOException {
...
+    @Test(expectedExceptions = NullPointerException.class)
+    public void testCheckWeakHashAlgorithmNull() {
+        PGPSignatureUtils.checkWeakHashAlgorithm(null);
+    }
+
+    @Test
+    public void testCheckWeakHashAlgorithmAllAlgorithms() {
+        PGPSignature sig = mock(PGPSignature.class);
+        // added `-1` as final response to detect when assertions are out-of-sync, as sort of safety net.
+        when(sig.getHashAlgorithm()).thenReturn(HashAlgorithmTags.MD5, HashAlgorithmTags.SHA1,
```
Comment (slawekjaranowski) [inline src/test/java/org/simplify4u/plugins/utils/PGPSignatureUtilsTest.java:50]: please use  @Test(dataProvider = "...") this is difficult to read

**Comment by cobratbq** [comment]
@slawekjaranowski can you verify if the Windows build-agents check out the repository with `core.autocrlf` set to `true` or `auto`? The `pom` file keeps failing verification and it works for me locally and on any posix-style build-agent.

I will revert the "experimental" changes to loading file content for verification as these weren't intended for this PR and only made to verify the build agent issues.

**Comment by slawekjaranowski** [comment]
We have no influence how users/ci system have configured `autocrlf`.
In this case we can create `.gitattributes` file and configure line end for specific files
https://git-scm.com/docs/gitattributes

File: src/it/resources/local-repo/nl/dannyvanheumen/helloworld/1.0-weaksignature/helloworld-1.0-weaksignature.pom [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                <artifactId>maven-clean-plugin</artifactId>
+                <version>2.5</version>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <version>2.6</version>
+            </plugin>
+        </plugins>
+    </build>
```
Comment (slawekjaranowski) [inline src/it/resources/local-repo/nl/dannyvanheumen/helloworld/1.0-weaksignature/helloworld-1.0-weaksignature.pom:106]: do we need such configurations for signature testing - maybe empty pom will be ok

File: .gitattributes [hunk 4]
```diff
@@ -0,0 +1 @@
+*.pom           text eol=lf
```
Comment (slawekjaranowski) [inline .gitattributes:1]: I will move this file to directory where we need specific line ends.
When we put in on project root - all poms will have lf but other text files in project can be changed by git ...
So project will have mixed line ends.

File: src/it/resources/local-repo/nl/dannyvanheumen/helloworld/1.0-weaksignature/helloworld-1.0-weaksignature.pom [hunk 5]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>nl.dannyvanheumen</groupId>
+    <artifactId>helloworld</artifactId>
+    <version>1.0-weaksignature</version>
+
```
Comment (slawekjaranowski) [inline src/it/resources/local-repo/nl/dannyvanheumen/helloworld/1.0-weaksignature/helloworld-1.0-weaksignature.pom:10]: Please change packaging to pom, and jar file can be removed.

**Comment by cobratbq** [comment]
@slawekjaranowski the PR is updated with a pom-only dependency with weak signature. Using only the pom is a nice catch. It makes a lot of things simpler. I completely missed that.
